sdk/go/common/util/executable/executable.go

```diff
@@ -26,7 +26,11 @@ func FindExecutable(program string) (string, error) {
 	}
 
 	cwdProgram := filepath.Join(cwd, program)
-	if fileInfo, err := os.Stat(cwdProgram); !os.IsNotExist(err) && !fileInfo.Mode().IsDir() {
+	fileInfo, err := os.Stat(cwdProgram)
+	if err != nil && !os.IsNotExist(err) {
+		return "", err
+	}
+	if err == nil && !fileInfo.Mode().IsDir() {
```

Comment:
Nit: Should we match line 52, which does the same thing but checks if fileInfo is not nil?

```
if fileInfo != nil && !fileInfo.Mode().IsDir() {
```

Reply:
Good point. They probably should do the same, I guess it's a question of if stat errors should be ignored (do like line 52), or returned as an error to the user (do like here).
Given the potential for confusing behaviour with silent errors I'd opted to return them here and if everyone agrees with that I can update line 52 (or someone else can as I'm only on my phone for a week)